pkg/storage/secret/metadata/keeper_store.go

```diff
@@ -102,46 +163,59 @@ func (s *keeperMetadataStorage) Update(ctx context.Context, newKeeper *secretv0a
 		return nil, fmt.Errorf("missing auth info in context")
 	}
 
-	currentRow := &keeperDB{Name: newKeeper.Name, Namespace: newKeeper.Namespace}
+	var oldKeeper *secretv0alpha1.Keeper
+	err := s.db.GetSqlxSession().WithTransaction(ctx, func(sess *session.SessionTx) error {
+		// Validate before updating that any `secureValues` referenced exist and do not reference other third-party keepers.\
 
-	err := s.db.WithTransactionalDbSession(ctx, func(sess *sqlstore.DBSession) error {
-		// Validate before updating that any `secureValues` referenced exist and do not reference other third-party keepers.
-		if err := s.validateSecureValueReferences(sess, newKeeper); err != nil {
+		if err := s.validateSecureValueReferences(ctx, sess, newKeeper); err != nil {
 			return err
 		}
 
-		found, err := sess.Get(currentRow)
+		var err error
+		oldKeeper, err = s.Read(ctx, xkube.Namespace(newKeeper.Namespace), newKeeper.Name)
 		if err != nil {
+			if errors.Is(err, contracts.ErrKeeperNotFound) {
+				return err
+			}
 			return fmt.Errorf("failed to get row: %w", err)
 		}
 
-		if !found {
-			return contracts.ErrKeeperNotFound
-		}
-
 		return nil
 	})
 	if err != nil {
 		return nil, fmt.Errorf("db failure: %w", err)
 	}
 
-	newRow, err := toKeeperUpdateRow(currentRow, newKeeper, authInfo.GetUID())
+	oldKeeperRow, err := toKeeperRow(oldKeeper)
+	if err != nil {
+		return nil, fmt.Errorf("failed to map to row: %w", err)
+	}
+
+	newRow, err := toKeeperUpdateRow(oldKeeperRow, newKeeper, authInfo.GetUID())
 	if err != nil {
 		return nil, fmt.Errorf("failed to map into update row: %w", err)
 	}
-	err = s.db.WithDbSession(ctx, func(sess *sqlstore.DBSession) error {
-		cond := &keeperDB{Name: newKeeper.Name, Namespace: newKeeper.Namespace}
 
-		if _, err := sess.Update(newRow, cond); err != nil {
+	req := &updateKeeper{
+		SQLTemplate: sqltemplate.New(s.dialect),
+		Row:         newRow,
+	}
+	q, err := sqltemplate.Execute(sqlKeeperUpdate, req)
+	if err != nil {
+		return nil, fmt.Errorf("update template %q: %w", q, err)
+	}
+
+	err = s.db.GetSqlxSession().WithTransaction(ctx, func(sess *session.SessionTx) error {
+		if _, err := sess.Exec(ctx, q, req.GetArgs()...); err != nil {
 			return fmt.Errorf("failed to update row: %w", err)
 		}
-
 		return nil
 	})
 	if err != nil {
 		return nil, fmt.Errorf("db failure: %w", err)
 	}
 
+	// TODO We are converting the new row(before the update operation) , should we query the db again??
```

Comment:
We could change the query to have a `RETURNING *`, and maybe use `sess.Get` or `sess.Query` but I haven't tested it 🤔

Reply:
This was more of a "philosophical" question, I don't think we will end up with the wrong values, it doesn't look like it. But we also don't have the real values in the db.